### PR TITLE
More syntax tests

### PR DESCRIPTION
**Pull request by etpinard**
This PR:

- makes the 'file name must be in lower case' syntax test run on (almost) all files
- add a new check that there's a `\n` at end of file for JSON and `.glsl` files
- update files that weren't adhering to these rules

### CODE REVIEW

File: tasks/test_syntax.js [hunk 1]
```diff
@@ -125,7 +127,39 @@ function assertFileNames() {
...
+}
+
+// check that all files have a trailing new line character
+function assertTrailingNewLine() {
+    var pattern = combineGlobs([
+        path.join(constants.pathToSrc, '**/*.glsl'),
+        path.join(constants.pathToRoot, 'test', 'image', 'mocks', '*')
+    ]);
+
+    var regex = /\r?\n$/;
```
Comment (alexcjohnson) [inline tasks/test_syntax.js]: do we want to make sure they end in exactly one newline, no more?

Comment (etpinard) [inline tasks/test_syntax.js]: Ah. Good call. Thanks!

Comment (etpinard) [inline tasks/test_syntax.js]: done in https://github.com/plotly/plotly.js/pull/1419/commits/f13091c5dd29ccadcb404d2a7c7975b7c1ec6341

**Comment by alexcjohnson** [comment]
Thanks, 💃 

Not sure what GH thinks changed in https://github.com/plotly/plotly.js/pull/1419/commits/9f1c6c4ee7a9bcda469564cee91d15a1232a7620#diff-ebde94117620b050ec349c85b9a200ea but anyway it all looks good.

**Pull request merged**